### ISSUE TITLE
Timeout not working with SHARED flag

### ISSUE THREAD

**Issue by daviddavo** [WoLpH/portalocker#74]
The timeout doesn't seem to work using the SHARED flag

I tried with this sample code (without the flag):
```python
from pathlib import Path
import os
from time import sleep
import portalocker as pl
from datetime import datetime

DIRECTORY = Path('.')
p_lock = Path(DIRECTORY / '.lock')
print("Start", datetime.now())
with pl.Lock(p_lock, 'r', timeout=5):
    print("Lock acquired", datetime.now())
    for i in range(10):
        print("Sleeping", i)
        sleep(1)
```

And issuing `flock .lock --command "sleep 120"` on another terminal, and as you can see on the following output, it raised an error after 5 seconds, just as expected:
```shell
$ flock .lock --command "sleep 120" & python sample.py
[1] 1019583
Start 2022-01-27 10:57:05.303649
Traceback (most recent call last):
  File "/home/davo/sample.py", line 10, in <module>
    with pl.Lock(p_lock, 'r', timeout=5):
  File "/home/davo/.local/lib/python3.10/site-packages/portalocker/utils.py", line 153, in __enter__
    return self.acquire()
  File "/home/davo/.local/lib/python3.10/site-packages/portalocker/utils.py", line 254, in acquire
    raise exceptions.LockException(exception)
portalocker.exceptions.LockException: [Errno 11] Resource temporarily unavailable
```

On the other hand, when I add the `SHARED` flag, it waits until the file is unlocked again
```python
from pathlib import Path
import os
from time import sleep
import portalocker as pl
from datetime import datetime

DIRECTORY = Path('.')
p_lock = Path(DIRECTORY / '.lock')
print("Start", datetime.now())
with pl.Lock(p_lock, 'r', timeout=5, flags=pl.LockFlags.SHARED):
    print("Lock acquired", datetime.now())
    for i in range(10):
        print("Sleeping", i)
        sleep(1)
```

As you can see in the output, it waits 15 seconds to access the file, instead of raising an error after 5
```shell
$  flock .lock --command "sleep 15" & python sample.py
[1] 1034452
Start 2022-01-27 11:07:03.311756
[1]  + 1034452 done       flock .lock --command "sleep 15"
Lock acquired 2022-01-27 11:07:18.264913
Sleeping 0
Sleeping 1
Sleeping 2
Sleeping 3
Sleeping 4
Sleeping 5
Sleeping 6
Sleeping 7
Sleeping 8
Sleeping 9
```

Btw, the docs should say that `timeout` is specified in seconds. I took me a while to detect the error because I thought the timeout was too big

**Comment by daviddavo** [comment]
It seems like `fcntl.flock` is what's causing the issue. It hangs in here, but it doesn't even let me pause the debugger.

https://github.com/WoLpH/portalocker/blob/b47d5f485f7c2e4d5af46307c6b3c4d2ea50e5f6/portalocker/portalocker.py#L138

**Comment by daviddavo** [comment]
Solution: I had to pass the `NON_BLOCKING` flag as well. Maybe it should be automtically added (or raise a warning) if `fail_when_locked` is True

**Comment by WoLpH** [comment]
I've done a bunch of debugging (I should have kept on reading, would have saved me time :P ) and `fail_when_locked` is not set in this case. And I don't think it should be.

The main issue is that when you're setting custom flags that you override the default so `NON_BLOCKING` isn't set either. I believe it is mostly a documentation issue because there are valid reasons to have a blocking lock without a timeout.

So... I think the best solution would be to add a warning if an explicit timeout is set when `NON_BLOCKING` is not set. That way you at least know what to expect :)

**Comment by WoLpH** [comment]
I've released version 2.5.0 which should fix the issues :)
